### PR TITLE
kernel: thread: move thread testcases of the tests/kernel/thread to new ZTEST api

### PR DESCRIPTION
**Pull request by Zhaoningx**
Move thread testcases of the tests/kernel/thread to new ZTEST api.

### CODE REVIEW

**Review by andyross — DISMISSED** [review]

File: arch/x86/core/fatal.c [hunk 1]
```diff
@@ -28,6 +28,7 @@ FUNC_NORETURN void arch_system_halt(unsigned int reason)
 	 * (often due to bad page tables or other CPU structures) it will
 	 * terminate with 0 error code.
 	 */
+	printk("arch_system_halt in X86\n");
```
Comment (dcpleung) [inline arch/x86/core/fatal.c]: Please remove in this commit.

**Review by dcpleung — CHANGES_REQUESTED** [review]

**Comment by Zhaoningx** [comment]
Hi @andyross @dcpleung, I have addressed the comment and done some changes for thread init test cases, could you help to review this PR again? Thanks.

By the way, I don't know what should I do for the CI building failure, have you seen this failure log? Thanks.

**Comment by dcpleung** [comment]
Errr.. the error said to increase `CONFIG_MAX_THREAD_BYTES`.

**Comment by enjiamai** [comment]
I saw another failure like this, anyone knows how to solve it? 

```
INFO    -  74/406 degu_evk                  tests/kernel/threads/thread_apis/kernel.threads.apis FAILED Build failure (build)
INFO    - /__w/zephyr/zephyr/twister-out/degu_evk/tests/kernel/threads/thread_apis/kernel.threads.apis/build.log

.....

-- Found assembler: /opt/toolchains/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:22 (message):
  CONFIG_USB_DEVICE_VID has default value 0x2FE3.

  This value is only for testing and MUST be configured for USB products.


CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:28 (message):
  CONFIG_USB_DEVICE_PID has default value 0x100.

  This value is only for testing and MUST be configured for USB products.
```

**Comment by Zhaoningx** [comment]
> Errr.. the error said to increase `CONFIG_MAX_THREAD_BYTES`.

Hi @dcpleung, thanks for your help, I have updated the PR.

**Review by enjiamai — APPROVED** [review]

**Comment by Zhaoningx** [comment]
> I saw another failure like this, anyone knows how to solve it?
> 
> ```
> INFO    -  74/406 degu_evk                  tests/kernel/threads/thread_apis/kernel.threads.apis FAILED Build failure (build)
> INFO    - /__w/zephyr/zephyr/twister-out/degu_evk/tests/kernel/threads/thread_apis/kernel.threads.apis/build.log
> 
> .....
> 
> -- Found assembler: /opt/toolchains/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
> CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:22 (message):
>   CONFIG_USB_DEVICE_VID has default value 0x2FE3.
> 
>   This value is only for testing and MUST be configured for USB products.
> 
> 
> CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:28 (message):
>   CONFIG_USB_DEVICE_PID has default value 0x100.
> 
>   This value is only for testing and MUST be configured for USB products.
> ```

Hi @enjiamai, it seems the CI failure has been addressed. Thanks.

**Review by dcpleung — APPROVED** [review]

**Pull request merged**